hack/runtime-migrator/cmd/restore/restore.go

```diff
@@ -86,14 +89,28 @@ func (r Restore) Do(ctx context.Context, runtimeIDs []string) error {
 			continue
 		}
 
+		if currentShoot.Generation == objectsToRestore.OriginalShoot.Generation {
```

Comment:
I'm not convinced if we should have this condition to skip restore process. What about restoring possibly deleted or broken CRBs? Those objects are are not related with shoot generation

Reply:
Done